src/server/rdb_test.cc

```diff
@@ -735,4 +735,19 @@ TEST_F(RdbTest, HugeKeyIssue4497) {
   EXPECT_EQ(Run({"flushall"}), "OK");
 }
 
+TEST_F(RdbTest, HugeKeyIssue4554) {
```

Comment:
This fails without my changes.

IMO, I wanted to reproduce via a replication test, because only replication is used on the issue. However, it proved to be a little more difficult than expected for reasons it's hard for me to explain in a few lines. I am fairly positive that this PR will address all the problems but I would like to have a binary asap to sent and verify that we did see the same/similar case on the issue as here.